DylibBinder/DBFunc.cs

```diff
@@ -0,0 +1,139 @@
+﻿using System;
+using SwiftReflector;
+using SwiftReflector.Demangling;
+using SwiftReflector.Inventory;
+using System.Collections.Generic;
+using Dynamo.SwiftLang;
+using SwiftReflector.TypeMapping;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace DylibBinder {
+	internal class DBFunc {
+		public DBFunc (TLFunction tlf, bool isMutating = false, bool isProtocol = false)
+		{
+			Name = tlf.Name.Name;
+			IsStatic = GetStaticStatus (tlf.Signature);
+			IsProperty = false;
+			HasThrows = tlf.Signature.CanThrow;
+			IsMutating = isMutating;
+			OperatorKind = tlf.Operator.ToString ();
+			HasInstance = !tlf.Signature.IsConstructor && !IsStatic;
+			// TODO not sure if/how to handle instances with protocols yet
+			if (isProtocol)
+				HasInstance = false;
```

Comment:
Not exactly, I set the HasInstance right before it with other criteria. It can not be protocol and still not HasInstance. I added the && !isProtocol to the assignment above though to clear up some stench!